pkg/api/core/shoot/warnings.go

```diff
@@ -60,6 +64,26 @@ func GetWarnings(_ context.Context, shoot, oldShoot *core.Shoot, credentialsRota
 	return warnings
 }
 
+func getWarningsForDeprecatedNodeLocalDNSLabels(shoot *core.Shoot) []string {
+	var (
+		warnings []string
+	)
+
+	if _, ok := shoot.Annotations[v1beta1constants.AnnotationNodeLocalDNS]; ok {
+		warnings = append(warnings, fmt.Sprintf("annotation %v is deprecated. Use field `.spec.systemComponents.nodeLocalDNS.enabled` in Shoot instead.", v1beta1constants.AnnotationNodeLocalDNS))
+	}
```

Comment:
@ScheererJ shared the following:
> Switching on node-local-dns via shoot specification will roll the nodes even if node-local-dns was enabled beforehand via annotation.

This is also present in https://github.com/gardener/gardener/issues/7581. So as written in the above form, it might be quite unexpected for end users to switch to the new way and to receive a Node rollout because of this.

I think we should give people much more time to adapt or look into how to make the migration graceful without Node rollout.

Reply:
@ialidzhikov, I have updated the dropping version to v1.78. This should be enough I guess. Now there 4 version gap 74-77, which will be around 8 weeks.

Reply:
If I am regular end user, I see this warning. How I am supposed to know that changing from the annotation to the label will cause me a Node rollout? You could describe this in the warning or link to docs that describe this...

Reply:
/hold

Reply:
/unhold
@ialidzhikov I checked if we could prevent the node rollout, but It seems there is no way to do it.
Worker group hash changes based on if `node-local-dns` is enabled in shoot spec or not, [ref](https://github.com/gardener/gardener/blob/5abed1bfedd6f2188e5dcec9af161d1b5c286236/extensions/pkg/controller/worker/machines.go#L169-L172).
To prevent node rollout we will have to drop [this](https://github.com/gardener/gardener/blob/5abed1bfedd6f2188e5dcec9af161d1b5c286236/extensions/pkg/controller/worker/machines.go#L169-L172) part of code, which can't be done based on what my understanding is.